src/components/PageBlocks/RecordListFilter.vue

```diff
@@ -0,0 +1,249 @@
+      required: true,
+      default: () => {},
+    },
+    recordListFilter: {
+      type: Array,
+      required: true,
+      default: () => [],
+    },
+  },
+  data () {
+    return {
+      selectedFieldName: this.selectedField.key,
+      componentFilter: [...this.recordListFilter],
+      equations: [
+        { value: 'Contains', text: 'Contains' },
+        { value: 'Equal', text: 'Equal' },
+        { value: 'Not equal', text: 'Not equal' },
+      ],
+      conditions: [
+        { value: 'AND', text: 'AND' },
+        { value: 'OR', text: 'OR' },
+      ],
+      inFilter: false,
+      checkbox: [
+        { value: true, text: 'True' },
+        { value: false, text: 'False' },
+      ],
+    }
+  },
+  watch: {
+    recordListFilter: {
+      handler: function () {
+        if (
+          this.recordListFilter.filter(x => x.name === this.selectedField.key)
+            .length > 0
+        ) {
+          this.inFilter = true
+        } else {
+          this.inFilter = false
+        }
+      },
+    },
+  },
+  methods: {
+    onChange (selected, index) {
+      const field = this.fields.filter(f => f.key === selected)[0]
+      const tempFilter = [...this.componentFilter]
+      this.moduleField = field.moduleField
+      tempFilter[index].options = field.moduleField.options.options
+      tempFilter[index].kind = field.moduleField.kind
+      tempFilter[index].moduleField = field.moduleField
+      this.componentFilter = [...tempFilter]
+    },
+    addFilter () {
+      this.componentFilter.push({
+        condition: 'AND',
+        name: this.selectedField.key,
+        equation: null,
+        query: null,
+        kind: 'String',
+        options: null,
+      })
+    },
+    deleteFilter (index) {
+      this.componentFilter.splice(index, 1)
+    },
+    onOpen () {
+      if (this.recordListFilter.length < 1) {
+        this.componentFilter.push({
+          condition: 'WHERE',
+          name: this.selectedField.key,
+          equation: null,
+          query: null,
+          kind: this.selectedField.moduleField.kind,
+          options: this.selectedField.moduleField.options.options,
+        })
+      } else {
+        this.componentFilter = [...this.recordListFilter]
+      }
+      this.$refs.tooltip.$emit('open')
+    },
+    save () {
+      if (this.componentFilter.some(e => e.equation === null)) {
+        alert('Please fill all equations')
+      }
+      this.$emit('filter', [...this.componentFilter])
+    },
+  },
+}
+</script>
+<style lang="scss">
+.ctooltip .tooltip-inner {
+  max-width: none !important;
+  padding: 20px;
+  color: #2d2d2d;
+  text-align: center;
+  background-color: inherit;
+  border-radius: 0.25rem;
+  background: white;
+  border-radius: 10px;
+  opacity: 1 !important;
+  box-shadow: 0px 3px 48px #00000026;
+}
+.ctooltip .arrow::before {
+  border-bottom-color: white;
+}
+.ctooltip {
+  opacity: 1 !important;
+}
+</style>
+<style lang="scss" scoped>
+.filter-row {
```

Comment:
Our policy is, if a bootstrap class exists for it, use it. 
In this case, this whole class can be replaced with 
class="d-flex px-2 w-100 mb-1"

Same for classes below, if bootstrap-class exists, use it in the component instead of defining new class